Doc/c-api/iter.rst

```diff
@@ -19,40 +20,41 @@ There are two functions specifically for working with iterators.
 
    .. versionadded:: 3.10
 
-.. c:function:: PyObject* PyIter_Next(PyObject *o)
-
-   Return the next value from the iterator *o*.  The object must be an iterator
-   according to :c:func:`PyIter_Check` (it is up to the caller to check this).
-   If there are no remaining values, returns ``NULL`` with no exception set.
-   If an error occurs while retrieving the item, returns ``NULL`` and passes
-   along the exception.
+.. c:function:: int PyIter_NextItem(PyObject *iter, PyObject **item)
 
-To write a loop which iterates over an iterator, the C code should look
-something like this::
+   Return ``1`` and set *item* to a :term:`strong reference` of the
+   next value of the iterator *iter* on success.
+   Return ``0`` and set *item* to ``NULL`` if there are no remaining values.
```

Comment:
Maybe mention StopIteration?

```suggestion
   Return ``0`` and set *item* to ``NULL`` if there are no remaining values (:exc:`StopIteration`).
```

Reply:
Since it's valid for an iterator to return NULL without setting a StopIteration exception, you should ignore my comment :-)